pest/src/position.rs

```diff
@@ -274,6 +274,25 @@ impl<'i> Position<'i> {
         false
     }
 
+    /// Matches the char at the `Position` against a filter function and returns `true` if a match
+    /// was made. If no match was made, returns `false` and `pos` will not be updated.
+    pub(crate) fn match_char_by<F>(&mut self, f: F) -> bool
```

Comment:
Might be worth annotating `#[inline]` for consistency. That said, I would have to do some profiling to see if there is a significant performance difference.

Reply:
I _think_ `#[inline]` doesn't have any effect for a `pub(crate) fn`. If I recall correctly, all `#[inline]` does is provide the required metadata cross-crate to inline the function. And as it's generic, that already happens anyway to provide monomorphization. It _might_ also hint LLVM that this is likely a good target for inlining, but LLVM is able to do whatever it pleases with that hint.

That said, consistency is a good thing, and the hint doesn't hurt, so I've gone ahead and added it.